marshmallow_dataclass/__init__.py

```diff
@@ -81,88 +82,192 @@ class User:
 # Recursion guard for class_schema()
 _RECURSION_GUARD = threading.local()
 
+if sys.version_info >= (3, 10):
+    # match_args, kw_only, slots
+
+    @overload
+    def dataclass(
+        _cls: Type[_U],
+        *,
+        repr: bool = True,
```

Comment:
While were messing around here, we should fix the typing of our `dataclass` to match [typeshed’s typing for `dataclasses.dataclass`](https://github.com/python/typeshed/blob/95444479acf928c0c91cb25e0c6b5a2f8ade25e7/stdlib/dataclasses.pyi#L55)

That allows for
```py
@dataclass
class A:
    ...
```

```py
@dataclass()
class B:
    ...
```

and

```py
@dataclass(**kwargs)
class C:
    ...
```

But does not allow for providing the class and keyword params all in one go. I.e. the following is *not supported* by `dataclasses.dataclass`:
```py
class D:
    ...

D = dataclass(D, frozen=True)
```